### PR TITLE
octane 5.5.2 beta enchantments

### PR DESCRIPTION
**Pull request by m-seldin**
- adding sonar coverage listener (https://issues.jenkins-ci.org/browse/JENKINS-53546)
- user story #678027 : Create function that converts the testToRun to aparameter of a job
- JIRA #52670: WorkflowGraph.FlowNodeContainer is causing memory performance issues
-defect #717008 : suite run that are triggered from Octane -stays "In Progress" state forever (JENKINS-53674)
- defect #718030: fixing SRF/SRL/PC tests type detection (JENKINS-53674) 
- issue-53873: handling internally any failure related to our own events processing

### CODE REVIEW

File: src/main/java/com/microfocus/application/automation/tools/octane/testrunner/VariableInjectionAction.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+ * © Copyright 2013 EntIT Software LLC
```
Comment (grount) [inline src/main/java/com/microfocus/application/automation/tools/octane/testrunner/VariableInjectionAction.java]: The correct license is without "© Copyright 2013 EntIT Software LLC"

File: src/main/java/com/microfocus/application/automation/tools/octane/testrunner/VariableInjectionAction.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import hudson.EnvVars;
+import hudson.model.AbstractBuild;
+import hudson.model.EnvironmentContributingAction;
+
+import javax.annotation.CheckForNull;
+import java.util.HashMap;
+import java.util.Map;
+
+public class VariableInjectionAction implements EnvironmentContributingAction {
```
Comment (grount) [inline src/main/java/com/microfocus/application/automation/tools/octane/testrunner/VariableInjectionAction.java]: Add documentation to the class

File: src/main/resources/com/hpe/application/automation/tools/run/SonarOctaneListener/config.jelly [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:s="/lib/samples">
+
+  <!--
```
Comment (grount) [inline src/main/resources/com/hpe/application/automation/tools/run/SonarOctaneListener/config.jelly]: Is there a need for this file?
If so remove commented out code and add the license.

File: src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/config.jelly [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<!--
+  ~ © Copyright 2013 EntIT Software LLC
```
Comment (grount) [inline src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/config.jelly]: The correct license is without "© Copyright 2013 EntIT Software LLC"

File: src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/help-framework.html [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<!--
+  ~ © Copyright 2013 EntIT Software LLC
```
Comment (grount) [inline src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/help-framework.html]: The correct license is without "© Copyright 2013 EntIT Software LLC"

File: src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/help.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<!--
+  ~ © Copyright 2013 EntIT Software LLC
```
Comment (grount) [inline src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/help.html]: Same

Comment (m-seldin) [reply src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/help.html]: fixed

File: src/main/resources/com/hpe/application/automation/tools/run/SonarOctaneListener/config.jelly (hunk 3 above)
Comment (m-seldin) [reply src/main/resources/com/hpe/application/automation/tools/run/SonarOctaneListener/config.jelly]: fixed

File: src/main/java/com/microfocus/application/automation/tools/octane/testrunner/VariableInjectionAction.java (hunk 2 above)
Comment (m-seldin) [reply src/main/java/com/microfocus/application/automation/tools/octane/testrunner/VariableInjectionAction.java]: fixed

**Comment by m-seldin** [comment]
@grount please see fixes.

File: src/main/java/com/microfocus/application/automation/tools/run/SonarOctaneListener.java [hunk 7]
```diff
@@ -0,0 +1,210 @@
+/*
```
Comment (grount) [inline src/main/java/com/microfocus/application/automation/tools/run/SonarOctaneListener.java]: Wrong license

File: src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/help.html (hunk 6 above)
Comment (grount) [inline src/main/resources/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder/help.html]: * Unresolved

**Comment by grount** [comment]
Approved for test build

**Pull request merged**